### ISSUE TITLE
npm run dev never finish to compile

### ISSUE THREAD

**Issue by arilishu** [vercel/next.js#66729]
### Link to the code that reproduces this issue

https://github.com/vercel/next-learn/tree/main/basics/learn-starter

### To Reproduce

This is happening everywhere on my computer, doesn't matter if it's an old or new project.
I just create a new folder, run:
```
npx create-next-app@latest
npm run dev
```
it will show:
```
dev
> next dev

  ▲ Next.js 14.2.3
  - Local:        http://localhost:3000

 ✓ Starting...
 ✓ Ready in 3.5s
```
And that's all, never move forward to the Compiling process
[Link with a video](https://www.loom.com/share/7d0646ab472e4dedb80d53e8bcc3bfd0)

### Current vs. Expected behavior

it should compile the site and leave it ready to run on localhost:3000

### Provide environment information

```bash
MAC Os Sonoma 14.5
Apple M2
npm 10.7.0
nvm 0.39.7
```

### Which area(s) are affected? (Select all that apply)

create-next-app, Developer Experience, Script (next/script)

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

This morning i uninstalled JAVA, and then reinstalled the latest version. That's all I can tell

**Comment by icyJoseph** [comment]
Hi,

The compiler is "lazy", so you have to actually hit an endpoint for it to process it. Go to a browser, or from the CLI cURL, to `localhost:3000` and that'll compile that page.

**Comment by palaklive** [comment]
I am encountering a same issue on a Mac system with the following configuration:

* Operating System: Mac OS Sonoma 14.5
* Node.js version: v20.14.0
* npm version: 10.7.0
* Next.js version: 14.2.4
* React version: 18.3.1
* React DOM version: 18.3.1

As a temporary workaround, I have found that adding the `--turbo` flag to the dev server resolves the issue. However, I believe a more robust solution is needed to address the root cause of the problem.

Furthermore, I am experiencing an issue with the build process, where it becomes stuck with the message `Creating an optimized production build...`.

@timneutkens, I would be grateful if you could assist with this issue. Thank you for your time and consideration."

**Comment by arilishu** [comment]
> Hi,
> 
> The compiler is "lazy", so you have to actually hit an endpoint for it to process it. Go to a browser, or from the CLI cURL, to `localhost:3000` and that'll compile that page.

You are right, Compiling process started when I hit the localhost endpoint (in my case port 4001). But now the process is stuck on the Compiling step:
This is what I see:
riellipschutz@Ariels-Laptop front % npm run dev

> todoflor@0.1.0 dev
> next dev -p 4001

  ▲ Next.js 14.2.3
  - Local:        http://localhost:4001
  - Environments: .env

 ✓ Starting...
 ✓ Ready in 1714ms
 ○ Compiling / ...


Any idea what's next?

**Comment by arilishu** [comment]
> Hi,
> 
> The compiler is "lazy", so you have to actually hit an endpoint for it to process it. Go to a browser, or from the CLI cURL, to `localhost:3000` and that'll compile that page.

After trying that approach it's working.
But it's taking long time to compile every single page.
here is an example every time I make an update the time it takes to compile:
 ✓ Compiled /favicon.ico in 2.7s (2693 modules)
 ✓ Compiled in 2.6s (2693 modules)
 ○ Compiling /product-detail/[id] ...
 ✓ Compiled /product-detail/[id] in 69.7s (2788 modules)
 ✓ Compiled in 4.5s (2773 modules)
 ✓ Compiled in 1933ms (2759 modules)
 ○ Compiling /login ...
 ✓ Compiled /login in 40.3s (2836 modules)

Any idea what could be wrong?

**Comment by palaklive** [comment]
I manually defined the port with `"dev": "next dev -p 3001"`, and it started working.

**Comment by arilishu** [comment]
> I manually defined the port with `"dev": "next dev -p 3001"`, and it started working.

The issue is not related to the port. I have tried lot of ports but the errors still the same.

**Comment by timneutkens** [comment]
Can you try running with Turbopack (`--turbopack`) on Next.js 15 and let me know if you're still running into issues?

**Comment by quynhchau-b** [comment]
I try it and run successed 
{
  "scripts": {
    "dev": "next dev --turbo",
    ...
  },
  ...
}

**Comment by Smef** [comment]
I've been experiencing something similar. Eventually it seemed to give up and threw an error about memory usage before restarting the server.

This is a very simple test site, and it looks like the page worked immediately after it restarted. The favicon, which is just the new project default, seems to have taken a while still, though.

Next 15.2.3
pnpm 10.6.5

```
 ○ Compiling / ...
 ✓ Compiled / in 38.2s
 ⚠ Server is approaching the used memory threshold, restarting...
   ▲ Next.js 15.2.3 (Turbopack)
   - Local:        http://localhost:3000
   - Network:      http://192.168.1.212:3000

 ✓ Starting...
 ✓ Ready in 745ms
 ○ Compiling / ...
 ✓ Compiled / in 875ms
 GET / 200 in 1081ms
 ○ Compiling /favicon.ico ...
 ✓ Compiled /favicon.ico in 9.3s
 GET /favicon.ico?favicon.45db1c09.ico 200 in 33171ms
```

**Comment by Smef** [comment]
I think this might be related to this issue with memory usage: https://github.com/vercel/next.js/issues/54708

**Comment by timneutkens** [comment]
Going to close this issue as it's not reproducible and the original author hasn't replied in a long time.
